providers/implementations/rands/crngt.c

```diff
@@ -92,6 +92,17 @@ static const OPENSSL_CTX_METHOD rand_crng_ossl_ctx_method = {
     rand_crng_ossl_ctx_free,
 };
 
+static int prov_crngt_compare_previous(const unsigned char *prev,
+                                        const unsigned char *cur,
+                                        size_t sz)
+{
+    const int res = memcmp(prev, cur, sz) != 0;
+
+    if (!res)
+        ossl_set_error_state();
```

Comment:
Should this be configurable?
 i.e on by default but able to be turned off..
Should it be able to be disabled at compile time?
I had to do similar things for #12745

Reply:
I don't think so.  This happens when the seed source for the entire RNG system fails which is not something that should be tolerated.  If it were made configurable, turning it off would be very dubious from the FIPS viewpoint and it isn't running outside of FIPS anyway so that's kind of moot.

The CRNGT is a pretty weak test of _working_ but the false positive rate is designed to be incredibly low.  I.e. if it fails, you almost certainly have a very bad problem.  Unconditional error state seems reasonable here.

The ken gen errors, could (should) be more flexible about error state.  They are more likely to occur and less likely to be so devastating if they magically come good later.  I.e. those tests are much stronger and would likely continue to detect an ongoing problem.

Reply:
you should rewrirte "if (!res)" as an explicit comparison against zero.

Reply:
`res` is a boolean in this context, we don't mandate a zero comparison for booleans.

Reply:
ah right, I missed the "!= 0" above, sorry.  i'll suggest "res" is a bad name.  :)  "same" maybe?

Reply:
Or put the `!= 0` at the start to make it more obvious.